### PR TITLE
ideabuck: intellij test configuration

### PR DESCRIPTION
**Pull request by korniltsev**
The patch adds "Buck test" intellij configuration which runs "buck test" and shows results in intellij's test framework form.

Before we review/merge it I have a few questions. I'l leave them as a comment to the source.

There a bunch of things I would love to do in follow-up patches:
- Make this test configuration debuggable
- "Jump to source" in test tree view.
- Create test configurations from context (right click on test classes/methods)
- Install configuration (with debug)

So please let me know if fb will be ready to consider accept them.

### CODE REVIEW

File: src/com/facebook/buck/intellij/ideabuck/src/com/facebook/buck/intellij/ideabuck/configurations/BuckToGeneralTestEventsConverter.java [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+
+  public BuckToGeneralTestEventsConverter(
+      @NotNull String testFrameworkName,
+      @NotNull TestConsoleProperties consoleProperties,
+      @NotNull ProcessHandler handler,
+      @NotNull Project project) {
+    super(testFrameworkName, consoleProperties);
+    myHandler = handler;
+    mProject = project;
+    mConnection = mProject.getMessageBus()
```
Comment (styurin) [inline src/com/facebook/buck/intellij/ideabuck/src/com/facebook/buck/intellij/ideabuck/configurations/BuckToGeneralTestEventsConverter.java]: Not sure it's a good idea to connect and subscribe in the constructor. Isn't `consumeTestRunStarted` a better place for that?

File: src/com/facebook/buck/intellij/ideabuck/src/com/facebook/buck/intellij/ideabuck/configurations/BuckToGeneralTestEventsConverter.java [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+  }
+
+  @Override
+  public void consumeTestRunStarted(long timestamp) {
+    final GeneralTestEventsProcessor processor = getProcessor();
+    if (processor == null) {
+      return;
+    }
+    processor.onUncapturedOutput(
+        "Test run started (" + timestamp + ").\n",
```
Comment (styurin) [inline src/com/facebook/buck/intellij/ideabuck/src/com/facebook/buck/intellij/ideabuck/configurations/BuckToGeneralTestEventsConverter.java]: timestamp doesn't look like valuable information here. Can you convert it to readable format?

File: src/com/facebook/buck/intellij/ideabuck/src/com/facebook/buck/intellij/ideabuck/configurations/TestConfigurationEditor.java [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+  @Override
+  protected void resetEditorFrom(@NotNull TestConfiguration s) {
+    mTarget.setText(s.data.target);
+    mAdditionalParams.setText(s.data.additionalParams);
+    mTestSelectors.setText(s.data.testSelectors);
+  }
+
+  @Override
+  protected void applyEditorTo(@NotNull TestConfiguration s) throws ConfigurationException {
```
Comment (styurin) [inline src/com/facebook/buck/intellij/ideabuck/src/com/facebook/buck/intellij/ideabuck/configurations/TestConfigurationEditor.java]: can you rename `s` to something meaningful?

File: src/com/facebook/buck/intellij/ideabuck/src/com/facebook/buck/intellij/ideabuck/configurations/TestConfigurationEditor.java [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+  protected JComponent createEditor() {
+    return root;
+  }
+
+  public TestConfigurationEditor() {
+    root = new JPanel(new GridBagLayout());
+    final JBLabel targetLabel = new JBLabel();
+    targetLabel.setText("Target");
+    mTarget = new JBTextField();
+    mTarget.getEmptyText().setText("May be empty");
```
Comment (styurin) [inline src/com/facebook/buck/intellij/ideabuck/src/com/facebook/buck/intellij/ideabuck/configurations/TestConfigurationEditor.java]: Better text would be: Leave empty to run all tests

**Comment by styurin** [comment]
Looks good, can you merge the second commit into the first one?

**Comment by styurin** [comment]
Please, address comment about `BuckToGeneralTestEventsConverter` constructor. Also, could you update `plugin.xml` with a brief description of this feature. You can put it under version 2.7.3 since it's not released yet.

**Comment by korniltsev** [comment]
rebased

**Comment by styurin** [comment]
I still see the old commit, did you push your rebased commit?

**Comment by korniltsev** [comment]
Yes I did. ➜  buck git:(stacktraces) git push -f korniltsev
Everything up-to-date
My branch merges with master with no conflicts.

**Comment by korniltsev** [comment]
Rebased again, on fresh master (25e90b5)

**Comment by styurin** [comment]
Which version of IntelliJ did you use to get these jar files?

**Comment by korniltsev** [comment]
This jars are from intellij 2016.2.1

**Comment by korniltsev** [comment]
Btw, could you please take a look at comments that I've left here on github.

**Comment by styurin** [comment]
> So please let me know if fb will be ready to consider accept them.

Definitely.

**Pull request closed without merging**